base/compiler/abstractinterpretation.jl

```diff
@@ -161,18 +178,22 @@ function abstract_call_gf_by_type(interp::AbstractInterpreter, @nospecialize(f),
             add_backedge!(edge::MethodInstance, sv)
         end
         fullmatch = false
-        for i in napplicable:-1:1
-            match = applicable[i]::SimpleVector
-            method = match[3]::Method
-            if atype <: method.sig
-                fullmatch = true
-                break
+        if length(mts) == 1
+            for i in napplicable:-1:1
+                match = applicable[i]::SimpleVector
+                method = match[3]::Method
+                if atype <: method.sig
```

Comment:
```suggestion
                if match[4]
```
(this is new, but might as well start using it)

Reply:
`match[4]` or `!match[4]`? Which way around is this boolean?

Reply:
I tried this change and it causes singificant sysimage build time regressions. Given that it is an orthogonal change to this PR, I'll drop this here and it can be in a separate PR where can debug why it's problematic for performance.

Reply:
Oh, it’s probably getting confused by Union-splitting. Carry on.